content/src/content/jcr_root/apps/core/wcm/components/download/v1/download/download.html

```diff
@@ -0,0 +1,34 @@
+<!--/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  ~ Copyright 2018 Adobe Systems Incorporated
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/-->
+<div data-sly-use.download="com.adobe.cq.wcm.core.components.models.Download"
+     data-sly-use.templates="core/wcm/components/commons/v1/templates.html"
+     data-sly-test.hasContent="${download.imagePath || download.title || download.description}"
+     class="cmp-download${!wcmmode.disabled ? ' cq-dd-file' : ''}">
+    <div class="cmp-download__image" data-sly-test.imagePath="${download.imagePath}">
+        <img src="${imagePath}"/>
+    </div>
+    <h2 class="cmp-download__title" data-sly-test.title="${download.title}" data-sly-element="${download.titleType}">
+        ${title}
+    </h2>
+    <div class="cmp-download__description" data-sly-test.description="${download.description}">
+        ${description @ context="html"}
+    </div>
+    <a class="cmp-download__action ${download.extension}" href="${download.downloadUrl}">${download.actionText}</a>
```

Comment:
@bpauli - we should probably ensure we have a consistent way of marking up icons cross-component, see for example the carousel actions, which use a separate span for the icons [0].

Additionally I think we should namespace this css class with a BEM prefix.

[0] - https://github.com/adobe/aem-core-wcm-components/blob/master/content/src/content/jcr_root/apps/core/wcm/components/carousel/v1/carousel/carousel.html#L39

Reply:
@richardhand As discussed offline I added the cmp-icon-- prefix to the extension.